### PR TITLE
otelmux: Treat incoming trace ID as links rather than parents

### PR DESCRIPTION
**Pull request by elaous**
When a trace is sent with a trace context creates a problem where the traces is trying to making our traces a child of a parent that is not sent to the collectors. The `otelmux` package that we use doesn't provide a solution to this problem but `otelhttp` does. Using `otelhttp` as a guide marking endpoints as public is a good way to avoid orphan spans. This will treat incoming trace ID as links rather than parents.

### CODE REVIEW

File: instrumentation/github.com/gorilla/mux/otelmux/mux.go
Comment (MrAlias) [inline instrumentation/github.com/gorilla/mux/otelmux/mux.go:59]: Setting this to a non-nil value and then checking below if this value is nil doesn't make sense. Why not just leave this as `nil` and have the guard below prevent the nil deref?

**Review by MrAlias — APPROVED** [review]

**Review by dmathieu — DISMISSED** [review]

**Comment by elaous** [comment]
Hi, just wondering, what are the next steps. There is anything on my side that needs to be done?

**Comment by dmathieu** [comment]
First, it looks like one of your commits is having CLA issues. Can you fix that?

**Comment by elaous** [comment]
> First, it looks like one of your commits is having CLA issues. Can you fix that?
I updated with changes from master and it fixed the issue.

**Comment by pellared** [comment]
@open-telemetry/go-maintainers This PR has already 3 approvals so it is probably good to merge.

**Comment by MrAlias** [comment]
> @open-telemetry/go-maintainers This PR has already 3 approvals so it is probably good to merge.

Reviews were re-requested for @Aneurysm9 and @dmathieu and they have not provided them.

**Comment by elaous** [comment]
> > @open-telemetry/go-maintainers This PR has already 3 approvals so it is probably good to merge.
> 
> Reviews were re-requested for @Aneurysm9 and @dmathieu and they have not provided them.
Sorry for the clueless question: How I dismissing a pull request review? Has been some time since I used Github.

**Comment by MrAlias** [comment]
> > > @open-telemetry/go-maintainers This PR has already 3 approvals so it is probably good to merge.
> > 
> > 
> > Reviews were re-requested for @Aneurysm9 and @dmathieu and they have not provided them.
> 
> Sorry for the clueless question: How I dismissing a pull request review? Has been some time since I used Github.

NP, I don't know if you can based on the project settings and the default permissions.

I can dismiss them if you like?

**Comment by elaous** [comment]
> > > > @open-telemetry/go-maintainers This PR has already 3 approvals so it is probably good to merge.
> > > 
> > > 
> > > Reviews were re-requested for @Aneurysm9 and @dmathieu and they have not provided them.
> > 
> > 
> > Sorry for the clueless question: How I dismissing a pull request review? Has been some time since I used Github.
> 
> NP, I don't know if you can based on the project settings and the default permissions.
> 
> I can dismiss them if you like?

Yes please.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by elaous** [comment]
There is anything I can do about codeconv/patch?

**Comment by MrAlias** [comment]
> There is anything I can do about codeconv/patch?

If you would like to submit a follow up PR to help improve the projects overall coverage we would be happy to have it. But, for this PR, the coverage is good enough.

**Comment by ash2k** [comment]
Would be great to have this for the gRPC instrumentation too.

**Pull request merged**
